fujiji-server/src/repositories/listing.js

```diff
@@ -50,6 +50,62 @@ async function getAllListingsByCityCondition(city, condition) {
   }
 }
 
+async function getListingsBySearch(searchParameters) {
+  try {
+    const conditions = [];
+    const values = [];
+
+    if (typeof searchParameters.title !== 'undefined') {
+      conditions.push('title LIKE ?');
+      values.push(`%${searchParameters.title}%`);
+    }
+
+    if (typeof searchParameters.condition !== 'undefined') {
+      conditions.push('condition = ?');
+      values.push(searchParameters.condition);
+    }
+
+    if (typeof searchParameters.category !== 'undefined') {
+      conditions.push('category = ?');
+      values.push(searchParameters.category);
+    }
+
+    if (typeof searchParameters.city !== 'undefined') {
+      conditions.push('city = ?');
+      values.push(searchParameters.city);
+    }
+
+    if (typeof searchParameters.province !== 'undefined') {
+      conditions.push('province_code = ?');
+      values.push(searchParameters.province);
+    }
+
+    if (typeof searchParameters.startPrice !== 'undefined' && typeof searchParameters.endPrice !== 'undefined') {
+      conditions.push('price BETWEEN ? AND ?');
+      values.push(parseInt(searchParameters.startPrice, 10));
+      values.push(parseInt(searchParameters.endPrice, 10));
+    }
+
+    const conditionsStr = {
+      where: conditions.length
+        ? conditions.join(' AND ') : '1',
+      values,
+    };
+
+    const [listingsBySearch] = await sequelize.query(
+      `SELECT * FROM fujiji_listing WHERE ${conditionsStr.where}`,
```

Comment:
I've just remembered this, can you also put `ORDER BY creation_date DESC` at the end? So that the search result is in reverse chronological order.

Reply:
I think we have an issue for this: https://github.com/devin-efendy/fujiji/issues/95

Reply:
Since it's a new implementation anyway, I just thought why not do it right from here? But don't mind fixing it after too.